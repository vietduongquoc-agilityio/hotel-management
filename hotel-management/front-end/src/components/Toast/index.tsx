import { useToast } from "@chakra-ui/react";

const Toast = () => {
  const toast = useToast();

  const showToast = (
    title: string,
    status: "info" | "warning" | "success" | "error"
  ) => {
    toast({
      title,
      status,
      duration: 3000,
      isClosable: true,
      position: "top",
    });
  };

  return { showToast };
};

export default Toast;




