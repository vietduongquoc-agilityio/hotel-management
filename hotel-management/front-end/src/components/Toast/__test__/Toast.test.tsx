import { renderHook, act } from "@testing-library/react";
import { useToast } from "@chakra-ui/react";
import Toast from "..";

jest.mock("@chakra-ui/react", () => ({
  useToast: jest.fn(),
}));

describe("Toast", () => {
  const mockUseToast = jest.fn();

  beforeEach(() => {
    (useToast as jest.Mock).mockImplementation(() => mockUseToast);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should call useToast with correct parameters for info toast", () => {
    const { result } = renderHook(() => Toast());

    act(() => {
      result.current.showToast("Info message", "info");
    });

    expect(result).toMatchSnapshot();
    expect(mockUseToast).toHaveBeenCalledWith({
      title: "Info message",
      status: "info",
      duration: 3000,
      isClosable: true,
      position: "top",
    });
  });

  it("should call useToast with correct parameters for success toast", () => {
    const { result } = renderHook(() => Toast());

    act(() => {
      result.current.showToast("Success message", "success");
    });

    expect(mockUseToast).toHaveBeenCalledWith({
      title: "Success message",
      status: "success",
      duration: 3000,
      isClosable: true,
      position: "top",
    });
  });

  it("should call useToast with correct parameters for warning toast", () => {
    const { result } = renderHook(() => Toast());

    act(() => {
      result.current.showToast("Warning message", "warning");
    });

    expect(mockUseToast).toHaveBeenCalledWith({
      title: "Warning message",
      status: "warning",
      duration: 3000,
      isClosable: true,
      position: "top",
    });
  });

  it("should call useToast with correct parameters for error toast", () => {
    const { result } = renderHook(() => Toast());

    act(() => {
      result.current.showToast("Error message", "error");
    });

    expect(mockUseToast).toHaveBeenCalledWith({
      title: "Error message",
      status: "error",
      duration: 3000,
      isClosable: true,
      position: "top",
    });
  });
});
