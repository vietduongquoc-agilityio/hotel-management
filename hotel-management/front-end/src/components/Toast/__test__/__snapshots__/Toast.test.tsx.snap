// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should call useToast with correct parameters for info toast 1`] = `
{
  "current": {
    "showToast": [Function],
  },
}
`;
